rpi_backlight.py

```diff
@@ -86,49 +102,73 @@ def set_brightness(value: int, smooth: bool = False, duration: float = 1) -> Non
         diff = abs(value - actual)
         while actual != value:
             actual = actual - 1 if actual > value else actual + 1
-
-            _set_value("brightness", actual)
+            set_brightness_value(actual)
             time.sleep(duration / diff)
     else:
-        _set_value("brightness", value)
+        set_brightness_value(value)
 
 
 def set_power(on: bool) -> None:
     """Set the display power on or off.
-
     :param on: Boolean whether the display should be powered on or not
     """
-
     # 0 is on, 1 is off
-    _set_value("bl_power", int(not on))
-
+    if MODE == "MODE_RPI":
+        _set_value("bl_power", int(not get_power))
+    elif MODE == "MODE_TINKERBOARD":
+        if on:
+            value = 255
+        else: 
+            value = 0
+        set_brightness_value(value)
+
+
+def toggle_power() -> None:
+    """Toggle the display power on or off."""
+    if MODE == "MODE_RPI":
```

Comment:
This whole function body should just say `set_power(not get_power())`, the necessary actions and mode checks already happen in `set_power`.

Reply:
Fixed